src/caffe/layers/base_data_layer.cpp

```diff
@@ -61,6 +61,9 @@ void BasePrefetchingDataLayer<Dtype>::Forward_cpu(
   // First, join the thread
   JoinPrefetchThread();
   DLOG(INFO) << "Thread joined";
+  // Reshape to loaded data.
+  top[0]->Reshape(1, this->prefetch_data_.channels(),
```

Comment:
Should this be `Reshape(this->prefetch_data_.num()`? This is unconditional, not just batch size one, right?